src/Avalonia.Controls.ColorPicker/ColorSpectrum/ColorSpectrum.cs

```diff
@@ -1121,27 +1112,20 @@ private async void CreateBitmapsAndColorMap()
             int pixelDimension = (int)Math.Round(minDimension * scale);
             var pixelCount = pixelDimension * pixelDimension;
             var pixelDataSize = pixelCount * 4;
-
-            bgraMinPixelData = new ArrayList<byte>(pixelDataSize);
-            bgraMaxPixelData = new ArrayList<byte>(pixelDataSize);
-            newHsvValues = new List<Hsv>(pixelCount);
-
             // We'll only save pixel data for the middle bitmaps if our third dimension is hue.
-            if (components == ColorSpectrumComponents.ValueSaturation ||
-                components == ColorSpectrumComponents.SaturationValue)
-            {
-                bgraMiddle1PixelData = new ArrayList<byte>(pixelDataSize);
-                bgraMiddle2PixelData = new ArrayList<byte>(pixelDataSize);
-                bgraMiddle3PixelData = new ArrayList<byte>(pixelDataSize);
-                bgraMiddle4PixelData = new ArrayList<byte>(pixelDataSize);
-            }
-            else
-            {
-                bgraMiddle1PixelData = new ArrayList<byte>(0);
-                bgraMiddle2PixelData = new ArrayList<byte>(0);
-                bgraMiddle3PixelData = new ArrayList<byte>(0);
-                bgraMiddle4PixelData = new ArrayList<byte>(0);
-            }
+            var middleBitmapsSize =
+                components is ColorSpectrumComponents.ValueSaturation or ColorSpectrumComponents.SaturationValue
+                    ? pixelDataSize : 0;
+
+            var newHsvValues = new List<Hsv>(pixelCount);
+            using var bgraMinPixelData = new PooledList<byte>(pixelDataSize, ClearMode.Never);
```

Comment:
Again, I'm not a fan of using a type for simple array data that requires disposing.

Reply:
Array pool data would require disposing either way, as array needs to be returned to the pool at some point.

Reply:
Well, I guess PooledList is overkill for this scenario. We only need an array with an .Add method for usability with existing code. Then we can re-use the same array reference as long as the bitmap size doesn't change. Just keep overwriting the data. No disposal needed.

I just don't like to see Dispose() much in high-level managed code... so it's just an observation. Then again, I also don't like the old ArrayList class either. Hopefully this all goes away in the future with composition brushes.